packages/block-library/src/post-template/index.php

```diff
@@ -19,14 +19,22 @@ function render_block_core_post_template( $attributes, $content, $block ) {
 	$page     = empty( $_GET[ $page_key ] ) ? 1 : (int) $_GET[ $page_key ];
 
 	$query_args = build_query_vars_from_query_block( $block, $page );
+
 	// Override the custom query with the global query if needed.
 	$use_global_query = ( isset( $block->context['query']['inherit'] ) && $block->context['query']['inherit'] );
 	if ( $use_global_query ) {
 		global $wp_query;
 		if ( $wp_query && isset( $wp_query->query_vars ) && is_array( $wp_query->query_vars ) ) {
 			// Unset `offset` because if is set, $wp_query overrides/ignores the paged parameter and breaks pagination.
 			unset( $query_args['offset'] );
-			$query_args = wp_parse_args( $wp_query->query_vars, $query_args );
+
+			// merge the user settings into the defaults of the query
+			$query_args = wp_parse_args( $query_args, $wp_query->query_vars );
```

Comment:
But it makes more sense, doesn't it to "merge the user settings into the defaults of the query" rather than the other way around? When you say:

> it breaks the whole query

can you instruct me how to, b/c testing seems to work as I expect it.